### ISSUE TITLE
Mouse-up event lost on raise window.

### ISSUE THREAD

**Issue by J-Dunn** [geany/geany#1159]
If I click and release on the blank, grey area to the right of the toolbar to raise the Geany window, it changes the cursor to the glyph of a fist and drags the window until I click again. 

It appears that the mouse up event is being lost. 

This does not happen if I click on the title bar, the menu or on the multidocument edit area or the grey blank area within the height of the tabs. It seems to be just the blank part of the toolbar. 

I'm using LXDE version of Fedora window manager behaviour configured for  'click to raise' and 'focus follows mouse'. 

Not a big issue but since I aim for the middle of the large, blank area without specific functionality to raise windows without setting of some other feature, I often get this annoying bug dragging my window around and I have to reposition it. 

This is the same behaviour as I would expect if I had not released the mouse, so I'm guessing that something is causing the mouse up event to be skipped without processing which would normally cancel the window dragging operation.

**Comment by elextr** [comment]
Doesn't happen here with Mint Cinnamon and the equivalent settings.  Here the fist occurs on mouse down on the titlebar, not on Geany toolbar, and thats the normal window managers action, not Geany's.  And it never loses the mouse up.

Since Geany never sets the fist icon it looks like a specific bug in LXDE.

**Comment by codebrainz** [comment]
Works OK with Xfwm. I get the drag cursor on mouse-press on empty menu or toolbar area as with other non-Geany windows, and it goes away (and stops dragging the window) when the mouse is released. If I had to guess, I'd say some window managers accept unhandled events from their child windows to make it easier to move the window around, some don't (whatever cinnamon uses, for ex.) and some have mouse-handling bugs (whatever lxde uses).

**Comment by J-Dunn** [comment]
Thanks for testing. 

It seems that the amount of overlap of the two windows is important in trigger this bug. If the other window is covering less than about 3/4 of the width of Geany it does not happen. 

It seems that it also needs to be deep enough to cover the status bar of Geany.

This is fairly typically how I set things up. I have other windows, like a file manager ( XFE ) and terminal windows ( LXTerminal ) which are about the same size as the Geany window and are offset by about 10% in both X and Y  so as to be able to access either, whichever one is on top.

It seems to depend in some way on what is getting redrawn. 

I initially thought this was a WM bug too but realised that I only ever see it happening when switching to Geany from another window. 

Since all this behaviour is handled by the WM it may be specific to Geany on LXDE

> Here the fist occurs on mouse down on the titlebar, not on Geany toolbar, and thats the normal window managers action

I do not see the bug on the menu bar, so this would not apply on a WM where that is the only drag area. 

 codebrainz , could you try again with two windows with a large amount of overlap as described? It seems that the drag behaviour on Xfwm is very similar to LXDE in that they provide drag on the black part of the menu and toolbar areas.

**Comment by elextr** [comment]
Well, since Geany has noting to do with window drag, if its happening with the cursor over parts of the Geany window then the WM is stealing the mouse clicks (deliberately or accidentally).

**Comment by elextr** [comment]
what versions of GTK and Glib? (near the top of menu->help->debug messages)

**Comment by codebrainz** [comment]
> could you try again with two windows with a large amount of overlap as described?

Works as expected.

```
$ xfwm4 --version
	This is xfwm4 version 4.12.3 (revision 7fdcb53) for Xfce 4.12
	Released under the terms of the GNU General Public License.
	Compiled against GTK+-2.24.30, using GTK+-2.24.30.

	Build configuration and supported features:
	- Startup notification support:                 Yes
	- XSync support:                                Yes
	- Render support:                               Yes
	- Xrandr support:                               Yes
	- Embedded compositor:                          Yes
	- KDE systray proxy (deprecated):               No
```

**Comment by J-Dunn** [comment]
```
10:33:42: Geany INFO		: Geany 1.27 (git >= 8afbc7a), en_GB.UTF-8
10:33:42: Geany INFO		: GTK 2.24.30, GLib 2.46.2
10:33:42: Geany INFO		: System data dir: /usr/local/share/geany
10:33:42: Geany INFO		: User config dir: /svn/stm32/.config/geany
10:33:42: Geany INFO		: System plugin path: /usr/local/lib/geany
10:33:42: Geany INFO		: Added filetype CUDA (61).
10:33:42: Geany INFO		: Added filetype JSON (62).
10:33:42: Geany INFO		: Added filetype Genie (63).
10:33:42: Geany INFO		: Added filetype Graphviz (64).
10:33:42: Geany INFO		: Added filetype Cython (65).
10:33:42: Geany INFO		: Added filetype Clojure (66).
10:33:42: Geany INFO		: Added filetype Scala (67).
```

**Comment by elextr** [comment]
@J-Dunn it appears t be something related to LXDE, perhaps ask on their forum.

**Comment by J-Dunn** [comment]
Yes, OK. Thanks for testing on similar but different platforms. Your comment about Geany not doing the drag and cursor changes makes sense.
